systemtest/src/main/java/io/strimzi/systemtest/utils/ClientUtils.java

```diff
@@ -51,20 +52,21 @@ public static void waitUntilClientReceivedMessagesTls(KafkaClientOperations kafk
     public static void waitTillContinuousClientsFinish(String producerName, String consumerName, String namespace, int messageCount) {
         LOGGER.info("Waiting till producer {} and consumer {} finish", producerName, consumerName);
         TestUtils.waitFor("continuous clients finished", Constants.GLOBAL_POLL_INTERVAL, timeoutForClientFinishJob(messageCount),
-            () -> kubeClient().getJobStatus(producerName) && kubeClient().getJobStatus(consumerName));
+            () -> kubeClient().checkSucceededJobStatus(producerName) && kubeClient().checkSucceededJobStatus(consumerName));
     }
 
     public static void waitForClientSuccess(String jobName, String namespace, int messageCount) {
         LOGGER.info("Waiting for producer/consumer:{} to finished", jobName);
         TestUtils.waitFor("job finished", Constants.GLOBAL_POLL_INTERVAL, timeoutForClientFinishJob(messageCount),
-            () -> kubeClient().namespace(namespace).getJobStatus(jobName));
+            () -> kubeClient().namespace(namespace).checkSucceededJobStatus(jobName));
     }
 
     public static void waitForClientTimeout(String jobName, String namespace, int messageCount) {
         LOGGER.info("Waiting for producer/consumer:{} to finish with failure.", jobName);
         try {
             TestUtils.waitFor("Job did not finish within time limit (as expected).", Constants.GLOBAL_POLL_INTERVAL, timeoutForClientFinishJob(messageCount),
-                () -> kubeClient().namespace(namespace).getJobStatus(jobName));
+                () -> kubeClient().namespace(namespace).checkSucceededJobStatus(jobName));
+            assertThat("Job did not finish with 1 failed pod (expected).", kubeClient().namespace(namespace).getJobStatus(jobName).getFailed().equals(1));
```

Comment:
We don't use assert in utils classes, so I think returning boolean and then asserting it in the test will be better approach. WDYT @Frawless @see-quick

Reply:
Yes, but we want to stop the tests at this point. Do you have any idea how to do that? It's out of waitFor.

Reply:
How we discussed, we can throw exception here and then, in tests, assert that exception is thrown.

Reply:
Still valid -> throw exception here and then assert it somewhere

Reply:
Added somehow appropriate `import java.rmi.UnexpectedException;`. If other Exception is needed, I can change it.